### PR TITLE
Image pull secrets

### LINKED ISSUE #270: Add `imagePullSecrets` to DaemonSets
Can the helm chart be expanded to enable `imagePullSecrets` for any required image pulls from a private repository? In order to pull images from a private repo today we have to export the template and manually add `imagePullSecrets` prior to each container stanza which is not ideal.

### PR DESCRIPTION
**Pull request by lkysow**
Allow setting image pull secrets for service accounts to enable pulling the consul-k8s and consul images from private registries.

Kube docs: https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#add-imagepullsecrets-to-a-service-account

Notes
* Refactors work from https://github.com/hashicorp/consul-helm/pull/402 (thanks!!)
* Closes https://github.com/hashicorp/consul-helm/pull/402
* Closes #270

### CODE REVIEW

File: values.yaml [hunk 1]
```diff
@@ -32,6 +32,16 @@ global:
...
   image: "consul:1.7.1"
 
+  # array of image pull secret names that will be applied to each service
+  # account. This can be used to reference image pull secrets if using
+  # a custom consul or consul-k8s Docker image.
+  # See https://kubernetes.io/docs/concepts/containers/images/#using-a-private-registry.
+  #
+  # Example:
+  #   imagePullSecrets:
+  #   - name: my-secret-name
```
Comment (adilyse) [inline values.yaml:42]: These example feels confusing. When empty, the value needs `[]`, but it doesn't need them when I give a value? In the [`extraConfig`](https://github.com/hashicorp/consul-helm/blob/master/values.yaml#L237) text we mention that it's an array of objects, which might help a bit.

**Review by adilyse — COMMENTED** [review]
Looking good. The one thing I noticed is that the value is defined as an array, but all of the tests only test a single value. It'd be good to make sure that the iteration through multiple values works as expected too.

File: values.yaml (hunk 1 above)
Comment (lkysow) [reply values.yaml]: Yeah good call, changed to say "array of objects containing image pull secret names"

**Comment by lkysow** [comment]
> The one thing I noticed is that the value is defined as an array, but all of the tests only test a single value. It'd be good to make sure that the iteration through multiple values works as expected too.

For sure. I've changed the tests to set two image pull secrets and assert that both are set.

**Review by adilyse — COMMENTED** [review]
Generally when we're testing for several values, each one is checked separately. Example: [env variable](https://github.com/hashicorp/consul-helm/blob/master/test/unit/server-statefulset.bats#L451-L475)

**Review by adilyse — COMMENTED** [review]
One more question-- should these be quoted? From the [docs](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#add-imagepullsecrets-to-a-service-account), it looks like they're not being quoted:

```
imagePullSecrets:
- name: myregistrykey
```

**Comment by lkysow** [comment]
It doesn't matter, it's just a string.

**Review by adilyse — APPROVED** [review]
Looks good.

Though if the quoting doesn't matter, it's a little odd to go to the effort to add it.

**Comment by lkysow** [comment]
Yeah I'll remove it.

**Pull request merged**